binary_classification/src/lib.rs

```diff
@@ -1,90 +1,156 @@
+        let hint = runtime_v1::supported_argument_type(ArgumentType::Float);
+        threshold.add_hint(&hint);
+        metadata.add_argument(&threshold);
+
+        let input = TensorMetadata::new("input");
+        input.set_description("The numbers to classify");
+        let hint =
+            supported_shapes(&[ElementType::F32], DimensionsParam::Fixed(&[0]));
+        input.add_hint(&hint);
+        metadata.add_input(&input);
+
+        let output = TensorMetadata::new("classified");
+        output.set_description("A tensor of `1`'s and `0`'s, where `1` indicates an element was above the `threshold` and `0` means it was below.");
+        let hint =
+            supported_shapes(&[ElementType::U32], DimensionsParam::Fixed(&[0]));
+        output.add_hint(&hint);
+        metadata.add_output(&output);
+
+        register_node(&metadata);
+    }
 
-impl BinaryClassification {
-    pub const fn new() -> Self { BinaryClassification { threshold: 0.5 } }
-}
+    fn graph(node_id: String) -> Result<(), GraphError> {
+        let ctx = GraphContext::for_node(&node_id)
+            .ok_or(GraphError::MissingContext)?;
+
+        ctx.add_input_tensor(
+            "input",
+            ElementType::U32,
+            DimensionsParam::Fixed(&[0]),
+        );
+        ctx.add_output_tensor(
+            "classified",
+            ElementType::U32,
+            DimensionsParam::Fixed(&[0]),
+        );
+
+        Ok(())
+    }
 
-impl Default for BinaryClassification {
-    fn default() -> Self { BinaryClassification::new() }
+    fn kernel(node_id: String) -> Result<(), KernelError> {
+        let ctx = KernelContext::for_node(&node_id)
+            .ok_or(KernelError::MissingContext)?;
+
+        let threshold = get_threshold(|n| ctx.get_argument(n))
+            .map_err(KernelError::InvalidArgument)?;
+
+        let TensorResult {
+            element_type,
+            dimensions,
+            buffer,
+        } = ctx.get_input_tensor("input").ok_or_else(|| {
+            KernelError::InvalidInput(InvalidInput {
+                name: "input".to_string(),
+                reason: BadInputReason::NotFound,
+            })
+        })?;
+
+        let output = match element_type {
+            ElementType::F32 =>{
+                buffer.view::<f32>(&dimensions)
+                .map_err(|e| KernelError::InvalidInput(InvalidInput{ name: "bounding_boxes".to_string(), reason: BadInputReason::InvalidValue(e.to_string()) }))?;
+                transform(buffer.elements(), threshold)
+            }
+            other => {
+                return Err(KernelError::Other(format!(
+                "The Object Filter proc-block doesn't support {:?} element type",
+                other,
+                )))
+            },
+        };
+
+        ctx.set_output_tensor(
+            "normalized",
+            TensorParam {
+                element_type: ElementType::U32,
+                dimensions: &dimensions,
+                buffer: &output.as_bytes(),
```

Comment:
FYI - the `as_bytes()` method already returns a `&[u8]`, so you don't need to add another reference.

```suggestion
                buffer: output.as_bytes(),
```